src/lib.rs

```diff
@@ -285,11 +285,27 @@ impl Iterator for MailEntries {
     }
 }
 
+
+/// Invalid subfolder name
+#[derive(Debug)]
+pub struct InvalidFolderName;
```

Comment:
This struct seems not too useful. I would prefer to just get rid of it. See other comment below